compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -239,26 +239,27 @@ impl<'f> PerFunctionContext<'f> {
                     return true;
                 }
 
-                let allocation_aliases_parameter =
-                    aliases.any(|alias| self.calls_reference_input.contains(&alias));
-                if allocation_aliases_parameter == Some(true) {
+                let allocation_aliases_instr_input =
+                    aliases.any(|alias| self.instruction_input_references.contains(&alias));
+                if allocation_aliases_instr_input == Some(true) {
```

Comment:
Probably unrelated to this PR, but we always check that the result of `AliasSet::any` is `Some(true)`. We could maybe change `AliasSet::any` to return a bool to simplify this code a bit.